RecoLocalCalo/EcalRecProducers/plugins/EcalRecHitProducer.cc

```diff
@@ -24,36 +24,51 @@
 #include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
 #include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
 
-EcalRecHitProducer::EcalRecHitProducer(const edm::ParameterSet& ps) {
-  ebRechitCollection_ = ps.getParameter<std::string>("EBrechitCollection");
-  eeRechitCollection_ = ps.getParameter<std::string>("EErechitCollection");
-
-  recoverEBIsolatedChannels_ = ps.getParameter<bool>("recoverEBIsolatedChannels");
-  recoverEEIsolatedChannels_ = ps.getParameter<bool>("recoverEEIsolatedChannels");
-  recoverEBVFE_ = ps.getParameter<bool>("recoverEBVFE");
-  recoverEEVFE_ = ps.getParameter<bool>("recoverEEVFE");
-  recoverEBFE_ = ps.getParameter<bool>("recoverEBFE");
-  recoverEEFE_ = ps.getParameter<bool>("recoverEEFE");
-  killDeadChannels_ = ps.getParameter<bool>("killDeadChannels");
-
+EcalRecHitProducer::EcalRecHitProducer(const edm::ParameterSet& ps)
+    : ebRechitCollection_(ps.getParameter<std::string>("EBrechitCollection")),
+      eeRechitCollection_(ps.getParameter<std::string>("EErechitCollection")),
+      doEB_(ps.getParameter<edm::InputTag>("EBuncalibRecHitCollection").label() != "none"),
```

Comment:
any reason to use a default InputTag of `"none"` over `""`?

Reply:
No particular reason. I think I have seen that before elsewhere. Maybe it is more explicit in the config file. But if `""` is the preferred way of doing it I can change.

Reply:
I have a preference for `""` over `"none"`, as the latter can have various forms with respect to capitalization, and the former seems slightly more common.

```
$ git grep InputTag | grep -i "cms.InputTag(\"\"" | wc -l
244
$ git grep InputTag | grep -i "cms.InputTag(\"none\"" | wc -l
191

```

Reply:
I will change it to `""` in that case. Maybe worth thinking about unifying this throughout CMSSW at some point.